rust/otel-arrow-rust/src/pdata/mod.rs

```diff
@@ -23,6 +23,18 @@ pub mod otlp;
 #[derive(Clone, Copy, Debug)]
 pub struct TraceID([u8; 16]);
 
+type Error = &'static str;
+
+impl<'a> TraceID {
+    pub fn try_new(value: &[u8]) -> Result<TraceID, Error> {
```

Comment:
You could enforce the length check at compile time by updating the method argument to an array or an array reference.

```rust
impl TraceID {
    pub fn new(value: &[u8; 16]) -> TraceID {
        TraceID(*value)
    }
}
```